### PR TITLE
Input data work

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address? #84 
Ref. #936

##### What are the significant changes in functionality due to this change request?
Converting code to use InputData.
This includes adding support for ParameterInput classes to have a check class that can see if a ParameterInput can parse a given xml node.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by joshua-cogliati-inl** [comment]
@alfoa  This is ready to review.

File: framework/Distributions.py [hunk 1]
```diff
@@ -763,16 +763,16 @@ class cls.
 
     return inputSpecification
 
-  def __init__(self):
+  def __init__(self, low=0.0, alpha=0.0, beta=1.0):
     """
       Constructor
       @ In, None
       @ Out, None
```
Comment (alfoa) [inline framework/Distributions.py:770]: can you update the docstring?

File: framework/utils/InputData.py [hunk 2]
```diff
@@ -35,8 +35,72 @@ class Quantity:
...
   one_to_infinity = (1,2)
 
-#
-#
+
+class CheckClass(object):
+  """
+  This checks to figure out if a node is a ParameterInput type
+  If the check passes, then it is.
+  """
```
Comment (alfoa) [inline framework/utils/InputData.py:43]:
```  
"""
    This checks to figure out if a node is a ParameterInput type
    If the check passes, then it is.
  """
```

File: framework/utils/InputData.py [hunk 3]
```diff
@@ -35,8 +35,72 @@ class Quantity:
...
+
+  def check(self, node):
+    """
+      Checks the node to see if it matches the checkDict
+      @ In, node, xml node to check
```
Comment (alfoa) [inline framework/utils/InputData.py:48]: the type is missing:

```@ In, node, **which type?**, xml node to check```

File: framework/utils/InputData.py [hunk 4]
```diff
@@ -35,8 +35,72 @@ class Quantity:
...
+      @ Out, bool, true if matches
+    """
+    assert False, "check function not implemented"
+
+
+  def failCheckReason(self, node):
+    """
+      returns a string about why the check failed
+      @ In, node, xml node to check
+      @ Out, string, message for user about why check failed.
```
Comment (alfoa) [inline framework/utils/InputData.py:58]: docstrings are not standard (same comments of above) .it is valid for all the others methods/docstrings

Comment (joshua-cogliati-inl) [reply framework/utils/InputData.py:58]: Fixed.

File: framework/utils/InputData.py (hunk 3 above)
Comment (joshua-cogliati-inl) [reply framework/utils/InputData.py]: Fixed.

File: framework/utils/InputData.py (hunk 2 above)
Comment (joshua-cogliati-inl) [reply framework/utils/InputData.py:43]: Fixed.

File: framework/Distributions.py (hunk 1 above)
Comment (joshua-cogliati-inl) [reply framework/Distributions.py:772]: Fixed.

**Comment by joshua-cogliati-inl** [comment]
@alfoa Ready to be reviewed again.

**Comment by alfoa** [comment]
Added reference to issue #936 (not closed yet)

**Comment by alfoa** [comment]
Checklist passed...

PR can be accepted...

**Pull request merged**